shelley/chain-and-ledger/formal-spec/utxo.tex

```diff
@@ -210,6 +210,14 @@ \subsection{UTxO Transitions}
     the amount consumed.
   \item
     The coin value of each new output must be non-negative.
+  \item
+    The transaction size must be below the allowed maximum.
+    Note that there is an implicit max transaction size given by the max block size,
+    and that if we wished to allow a transaction to be as large as will fit in a block, this
+    check would not be needed.
+    Though we do not have a concrete justification for restricting the size of a transaction
+    to be below that of the block, in the past this restriction has been helpful as a general
+    tunable parameter.
```

Comment:
> Though we do not have a concrete justification for restricting the size of a transaction
￼     to be below that of the block, in the past this restriction has been helpful as a general
￼     tunable parameter.

I wonder whether people would frown when seeing the sentence above... I don't want to be annoying, but I think it'd be nice if we had a more founded justification. I thought we wanted to bound the maximum transaction size to make sure we won't have very large transactions hanging in the mempool forever :man_shrugging: I don't know whether Philipp or Duncan gave additional reasons also.